astropy/nddata/ccddata.py

```diff
@@ -416,7 +416,11 @@ def _generate_wcs_and_update_header(hdr):
     wcs_header = wcs.to_header(relax=True)
     for k in wcs_header:
         if k not in _KEEP_THESE_KEYWORDS_IN_HEADER:
-            del new_hdr[k]
+            try:
+                del new_hdr[k]
+            except KeyError:
+                # Keyword does not exist in the original header
+                pass
```

Comment:
I don't know what the goal is but the code removes all WCS keywords except those three (as far as I can tell the CCDData class takes a header and a WCS object as two separate parameters). The problem here is how to decide what the WCS keywords are. The approach here (which I agree with) is to serialize the WCS object to a header. The problem is that this header is usually more complete than the original one and some of the keywords are missing from the original header.